persistence/remote/src/test/java/org/infinispan/persistence/remote/RemoteStoreSSLTest.java

```diff
@@ -0,0 +1,131 @@
+      gcr.registerComponent(timeService, TimeService.class);
+      gcr.rewire();
+      localCacheManager.getCache(REMOTE_CACHE).getAdvancedCache().getComponentRegistry().rewire();
+      ClassLoader cl = RemoteStoreSSLTest.class.getClassLoader();
+      SimpleServerAuthenticationProvider sap = new SimpleServerAuthenticationProvider();
+      HotRodServerConfigurationBuilder serverBuilder = HotRodTestingUtil.getDefaultHotRodConfiguration();
+      serverBuilder
+            .ssl()
+            .enable()
+            .requireClientAuth(true)
+            .keyStoreFileName(cl.getResource("keystore_server.jks").getPath())
+            .keyStorePassword("secret".toCharArray())
+            .keyAlias("hotrod")
+            .trustStoreFileName(cl.getResource("ca.jks").getPath())
+            .trustStorePassword("secret".toCharArray());
+      serverBuilder
+            .authentication()
+            .enable()
+            .serverName("localhost")
+            .addAllowedMech("EXTERNAL")
+            .serverAuthenticationProvider(sap);
+      hrServer = new HotRodServer();
+      hrServer.start(serverBuilder.build(), localCacheManager);
+
+      ConfigurationBuilder builder = TestCacheManagerFactory.getDefaultCacheConfiguration(false);
+      SecurityConfigurationBuilder remoteSecurity = builder
+            .persistence()
+            .addStore(RemoteStoreConfigurationBuilder.class)
+            .remoteCacheName(REMOTE_CACHE)
+            .remoteSecurity();
+      remoteSecurity
+            .ssl().enable()
+            .keyStoreFileName(cl.getResource("keystore_client.jks").getPath())
+            .keyStorePassword("secret".toCharArray())
+            .trustStoreFileName(cl.getResource("ca.jks").getPath())
+            .trustStorePassword("secret".toCharArray())
+            .addServer()
+            .host(hrServer.getHost())
+            .port(hrServer.getPort());
+      remoteSecurity
+            .authentication().enable()
+            .saslMechanism("EXTERNAL");
+      RemoteStore remoteStore = new RemoteStore();
+      remoteStore.init(createContext(builder.build()));
+
+      return remoteStore;
+   }
+
+   @Override
+   protected StreamingMarshaller getMarshaller() {
+      return localCacheManager.getCache(REMOTE_CACHE).getAdvancedCache().getComponentRegistry().getCacheMarshaller();
+   }
+
+   @Override
+   @AfterMethod
+   public void tearDown() {
+      HotRodClientTestingUtil.killServers(hrServer);
+      TestingUtil.killCacheManagers(localCacheManager);
+   }
+
+   @Override
+   protected boolean storePurgesAllExpired() {
+      return false;
+   }
+
+   @Override
+   public void testLoadAll() throws PersistenceException {
+   }
+
+   @Override
+   public void testReplaceExpiredEntry() throws Exception {
+      cl.write(marshalledEntry(internalCacheEntry("k1", "v1", 100l)));
+      // Hot Rod does not support milliseconds, so 100ms is rounded to the nearest second,
```

Comment:
It does support it http://infinispan.org/docs/dev/user_guide/user_guide.html#hot_rod_protocol_2_2, not sure how practical is to test that :)